Bonsai.Core/Expressions/IncludeWorkflowBuilder.cs

```diff
@@ -381,29 +387,26 @@ void EnsureWorkflow(IBuildContext buildContext)
                 var lastWriteTime = embeddedResource ? DateTime.MaxValue : File.GetLastWriteTime(path);
                 if (workflow == null || lastWriteTime > writeTime)
                 {
-                    var properties = workflow != null ? GetXmlProperties() : InternalXmlProperties;
+                    WorkflowBuilder builder;
+                    var properties = workflow != null ? GetXmlProperties(workflow) : InternalXmlProperties;
                     using (var stream = GetWorkflowStream(path, embeddedResource))
                     using (var reader = XmlReader.Create(stream, null, baseUri))
                     {
                         reader.MoveToContent();
                         var serializer = new XmlSerializer(typeof(WorkflowBuilder), reader.NamespaceURI);
-                        var builder = (WorkflowBuilder)serializer.Deserialize(reader);
-                        description = builder.Description;
-                        workflow = builder.Workflow;
-                        writeTime = lastWriteTime;
-                    }
-
-                    var parameterCount = workflow.GetNestedParameters().Count();
-                    SetArgumentRange(0, parameterCount);
-                    if (inspectWorkflow)
-                    {
-                        workflow = workflow.ToInspectableGraph();
+                        builder = (WorkflowBuilder)serializer.Deserialize(reader);
                     }
 
                     if (properties != null)
                     {
-                        SetXmlProperties(properties);
+                        SetXmlProperties(builder.Workflow, properties);
                     }
+
+                    var parameterCount = builder.Workflow.GetNestedParameters().Count();
+                    workflow = inspectWorkflow ? builder.Workflow.ToInspectableGraph() : builder.Workflow;
```

Comment:
Nothing following this is likely to throw, but this should still really be the very last thing assigned since it's the sentry value.

Reply:
The idea is that we shouldn't have to think about it this much in the first place, but fair enough with the argument range.

Ideally this should all be encapsulated better so that the object can't even exist in an invalid state at all, but that'd be a pretty major refactoring that's way out of scope for this PR.